test/test_networking.py

```diff
@@ -889,6 +889,33 @@ def mock_read(*args, **kwargs):
 
         assert exc_info.type is expected
 
+    def test_close(self, handler, monkeypatch):
+        rh = handler()
+        session = rh._get_instance(cookiejar=rh.cookiejar)
+        called = False
+        original_close = session.close
+
+        def mock_close(*args, **kwargs):
+            nonlocal called
+            called = True
+            return original_close(*args, **kwargs)
+
+        monkeypatch.setattr(session, 'close', mock_close)
+
+        rh.close()
+
+        assert called is True
```

Comment:
I think this is enough, especially since the values come from us. Same for the other ones
```suggestion
        assert called
```